components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -4824,57 +4795,87 @@ subroutine mam_pcarbon_aging_1subarea(                        &
-         if ( (iaer <= nsoa    ) .or. &
-              (iaer == iaer_nh4) .or. &
-              (iaer == iaer_no3) .or. &
-              (iaer == iaer_cl ) ) then
-            continue
-         else if (iaer == iaer_ncl) then
-            if (aging_include_seasalt .eqv. .false.) cycle
+         if (lmap_aer(iaer,nfrm) > 0) then
+            ! species is pom or bc
+            ! transfer the aged fraction to accum mode
+            ! include this transfer change in the cond and/or coag change (for mass budget)
+            call transfer_aged_pcarbon_mass_to_accum(iaer, nfrm, ntoo, &
+                                                     xferfrac_pcage, frac_cond, frac_coag, &
+                                                     qaer_cur, qaer_del_cond, qaer_del_coag)
          else
-            cycle
-         end if
+            ! species is soa, so4, or nh4 produced by condensation or coagulation
+            ! transfer all of it to accum mode
+            ! also transfer the condensation and coagulation changes
+            !    to accum mode (for mass budget)
+            call transfer_cond_coag_mass_to_accum(iaer, nfrm, ntoo, &
+                                                  qaer_cur, qaer_del_cond, qaer_del_coag)
+         endif
+      enddo
+      ! number - transfer the aged fraction to accum mode
+      ! include this transfer change in the cond and/or coag change (for mass budget)
+      call transfer_aged_pcarbon_num_to_accum(nfrm, ntoo, &
+                                              xferfrac_pcage, frac_cond, frac_coag, &
+                                              qnum_cur, qnum_del_cond, qnum_del_coag)
 
-         if ( (iaer == iaer_cl                    ) .and. &
-              (aging_include_seasalt .eqv. .false.) ) then
-            ! special case - only include the cl from condensation
-            tmp1 = max( qaer_del_cond(iaer,nfrm), 0.0_r8 )
-            tmp2 = max( qaer_del_coag_in(iaer,ipair), 0.0_r8 ) + tmp1
-            if (tmp2 >= 1.0e-35_r8) then
-               vol_shell = vol_shell + qaer_cur(iaer,nfrm)*fac_m2v_eqvhyg_aer(iaer)*(tmp1/tmp2)
-               tmp3 = tmp3 + qaer_del_cond(iaer,nfrm)*fac_m2v_eqvhyg_aer(iaer)
-            end if
-         else
-            vol_shell = vol_shell + qaer_cur(iaer,nfrm)*fac_m2v_eqvhyg_aer(iaer)
-            tmp3 = tmp3 + qaer_del_cond(iaer,nfrm)*fac_m2v_eqvhyg_aer(iaer)
-            tmp4 = tmp4 + qaer_del_coag_in(iaer,ipair)*fac_m2v_eqvhyg_aer(iaer)
-         end if
-      end do
+      enddo agepair_loop1
 
-      if ( do_cond ) then
-         tmp3 = max( tmp3, 1.0e-35_r8 )
-         tmp3 = tmp3/(tmp3 + max( tmp4, 0.0_r8 ))
-      else
-         tmp3 = 0.0_r8
-      end if
-      tmp4 = 1.0_r8 - tmp3
+      return
+      end subroutine mam_pcarbon_aging_1subarea
+
+
+      subroutine mam_pcarbon_aging_frac(nfrm, ipair, &
+          dgn_a, qaer_cur, qaer_del_cond, qaer_del_coag_in, &
+          xferfrac_pcage, frac_cond, frac_coag)      
+
```

Comment:
Add a short description of what this subroutine is doing. Same for other new subroutines.

Reply:
description added